shm/include/store.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "cargo.hpp"
+#include "player.hpp"
+#include "time.hpp"
+
+enum class Response {
+    done,
+    lack_of_money,
+    lack_of_cargo,
+    lack_of_space
+};
+
+class Store {
+    Store();
+    Response buy(Cargo* cargo, size_t amount, Player* player);
+    Response sell(Cargo* cargo, size_t amount, Player* player);
+    void nextDay() override;
```

Comment:
If you want to override nextDay(), you have to write `class Store : Observer {`.
See at the very beginning of the time.hpp to learn how to use Time and Observer classes.